Box.V2/Utility/Helper.cs

```diff
@@ -48,5 +50,59 @@ public static string Base64Encode(string plainText)
                     pair => Uri.UnescapeDataString(pair[ 1 ]));
         }
 
+        /// <summary>
+        /// Retuns the number of parts to be uploaded in Session
+        /// </summary>
+        /// <param name="totalSize"></param>
+        /// <param name="partSize"></param>
+        /// <returns></returns>
+        /// <exception cref="Exception"></exception>
+        public static int GetNumberOfParts(long totalSize, long partSize)
+        {
+            if (partSize == 0)
+                throw new Exception("Part Size cannot be 0");
+            int numberOfParts = 1;
+            if (partSize != totalSize)
+            {
+                numberOfParts = Convert.ToInt32(totalSize / partSize);
+                numberOfParts += 1;
+            }
+            return numberOfParts;
+        }
+
+        /// <summary>
+        /// Return a Random (probably unique) string.
+        /// </summary>
+        /// <param name="length">Lenght of the string returned.</param>
+        /// <returns></returns>
+        public static string GetRandomString(int length)
+        {
+            Random random = new Random();
+            byte[] buffer = new byte[length / 2];
+            random.NextBytes(buffer);
+            string result = String.Concat(buffer.Select(x => x.ToString("X2")).ToArray());
+            if (length % 2 == 0)
+                return result;
+            return result + random.Next(16).ToString("X");
+        }
+
+        /// <summary>
+        /// Returns part of the file starting at offset.
+        /// </summary>
+        /// <param name="stream"></param>
+        /// <param name="partSize"></param>
+        /// <param name="partOffset"></param>
+        /// <returns></returns>
+        public static Stream GetFilePart(Stream stream, long partSize, long partOffset)
```

Comment:
Not sure if I like all these utility functions belong to FilePartUpload as public here, should  be protected/private to FilesManager IMO, let's chat